src/main/frontend/worker/pipeline.cljs

```diff
@@ -119,8 +120,11 @@ default = false")
 (defn- invoke-hooks-default [repo conn {:keys [tx-meta] :as tx-report} context]
   (try
     (let [display-blocks-tx-data (add-missing-properties-to-typed-display-blocks (:db-after tx-report) (:tx-data tx-report))
-          tx-report* (if (seq display-blocks-tx-data)
-                       (let [result (ldb/transact! conn display-blocks-tx-data {:pipeline-replace? true})]
+          commands-tx (commands/run-commands tx-report)
```

Comment:
Do we need to run these commands in sync-way?
what about a background-task?

Reply:
Synchronous ensures that commands tx-data will be included in Undo/Redo history.
Currently, I'm thinking of executing tx-data synchronously but async for effects like summarizing a pdf.